print/libpaper/Portfile

```
@@ -1,55 +1,62 @@
 
 PortSystem      1.0
+PortGroup       github 1.0
+
+github.setup    rrthomas libpaper 2.1.2 v
+revision        0
+checksums       rmd160  6b3898d28155228927fa78f55bc747fbcd8e0bc2 \
+                sha256  1fda0cf64efa46b9684a4ccc17df4386c4cc83254805419222c064bf62ea001f \
+                size    1264200
+
+github.tarball_from  releases
 
-name            libpaper
-set mainversion 1.1.29
-# Keep in case need again in the future
-#set nmuversion  5
-#set realversion ${mainversion}+nmu${nmuversion}
-#version         ${mainversion}.${nmuversion}
-set realversion ${mainversion}
-version         ${mainversion}
 categories      print
-license         GPL-2
+# libpaper is LGPLv2.1+, paperspecs is public domain
+license         LGPL-2.1+ public-domain
 maintainers     nomaintainer
-platforms       darwin
 description     A library providing routines for paper size management
 long_description \
                 The paper library and accompanying files are intended \
                 to provide a simple way for applications to take \
                 actions based on a system- or user-specified paper size.
 
-homepage        https://packages.qa.debian.org/libp/libpaper.html
-master_sites    debian:libp/${name}/
-
-distname        ${name}_${realversion}
-worksrcdir      ${name}-${realversion}
-
-checksums       rmd160  8ab9f62014375cc307d3b3a21b813af030032e11 \
-                sha256  26330e21e9a3124658d515fd850b0cde546ff42d89b2596a5264c5f1677f0547 \
-                size    44942
-
-post-patch {
-    set fd [open [file join ${worksrcpath}/lib paperspecs] a 0644]
-    # additional paper sizes
-    foreach {n x y} {jisb0 2920 4127
-                     jisb1 2064 2920
-                     jisb2 1460 2064
-                     jisb3 1032 1460
-                     jisb4  729 1032
-                     jisb5  516  729
-                     jisb6  363  516} {
-        puts ${fd} "${n} ${x} ${y}"
-    }
-    close ${fd}
+subport libpaper-utils {
+    # We introduce a subport for files with different licenses.
+    # Including them in the 'libpaper' port would prevent the
+    # distribution of MacPorts binary archives for some prominent
+    # tools like Ghostscript or ImagicMagick.
+    description       The executables of the libpaper package
+    long_description  Provide the new 'paper' and deprecated 'paperconf' \
+                      utility programs that come with the libpaper \
+                      library. Both are used to print paper size \
+                      information.
+    # `paper` is GPL-3+, `paperconf` is GPL-2
+    license           GPL-3+ GPL-2
 }
 
 use_autoreconf  yes
 
-configure.args  --mandir=${prefix}/share/man    \
-                --sysconfdir=${prefix}/etc
+configure.args  --enable-relocatable
+
+if {${subport} eq ${name}} {
+    test.run     yes
+    test.target  check
 
-livecheck.type      regex
-livecheck.version   ${realversion}
-livecheck.regex     {"stable">([^<]+)}
+    post-destroot {
```

This change allows ghostscript to build, although I am not certain it is the correct approach:

```suggestion
    post-destroot {
            system "install_name_tool -id ${prefix}/lib/libpaper.2.dylib ${destroot}${prefix}/lib/libpaper.2.dylib"
           
```

---

Thanks, will do that soon.  Honestly, I think this is not right, or rather, MacPorts should be able to take care of it automatically – AFAICS, 'libpaper' properly supports the (more or less standardized) `$(DESTDIR)` variable, which is approximately the same as MacPort's `${destroot}` construction.

Looking into the MacPorts git repository, however, I see a surprising large number of packages doing the same...

---

Yes, any build system that needs its libs fixed manually with install_name_tool is badly behaved, but libtool does know how to do it right. This isn't anything to do with DESTDIR, rather it's about the install_name set when linking. It seems to be using `@rpath` but we normally prefer to use an absolute path (for just such reasons as the ghostscript problem).

Is this what the `--enable-relocatable` configure option controls? If it's not that, maybe running autoreconf without `--force` leaves something important un-regenerated? (Is autoreconf even still necessary with this new version?)

---

To answer my own question, yes, `--enable-relocatable` is causing the rpath use. The lib seems to have built OK locally without that and without use_autoreconf. I'm inclined to just remove both, unless you know a reason not to?

---

The only reason for using `--enable-relocatable` is to run the tests, which need this option.  If it's ok to not run the tests it can be removed.

I've added `use_autoreconf` by looking at other ports, so again, it can be easily removed if it is (no longer) necessary.

---

Hm, an unfortunate tradeoff with the tests. Presumably could be fixed by setting DYLD_LIBRARY_PATH, but that's not something I'm going to spend time digging into.

Running autoreconf is only necessary if either you patched one or more autotools input files, or the versions of autotools used to generate the distributed files was too old and doesn't work correctly on current systems.

---

Well, me just being a very, very casual Mac user, please proceed as you think it's best :slightly_smiling_face: